### ISSUE TITLE
CA1806 should warn if the method result is assigned to _

### ISSUE THREAD

**Issue by Bosch-Eli-Black** [dotnet/roslyn-analyzers#5823]
### Analyzer

**Diagnostic ID**: [CA1806](https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1806)

### Describe the improvement

This code warns:

```c#
int.TryParse("asd", out int result);
```

This code doesn't warn:

```c#
_ = int.TryParse("asd", out int result);
```

IMO, assigning the result of `TryParse()` to `_` is the same as ignoring it, so the second case should also warn.

### Additional context

I discovered this by watching people convert `int.TryParse("asd", out int result);` to `_ = int.TryParse("asd", out int result);` to get rid of the CA1806 warning 🙂

**Comment by Youssef1313** [comment]
I think this is by design. By using the discard you say "I intentionally don't want the method return value"

**Comment by mavasani** [comment]
Agreed. The discard is equivalent to an intentional suppression in source of an unused value.
